src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1038,7 +1038,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
 
         BuildData previousBuildData = getBuildData(build.getPreviousBuild());   // read only
         BuildData buildData = copyBuildData(build.getPreviousBuild());
-        build.addAction(buildData);
```

Comment:
FYI PR plugins doing trick around this buildData, but this may be related to `getBuildData` without build chooser...